osu.Game.Tests/Visual/Multiplayer/TestSceneGameplayChatDisplay.cs

```diff
@@ -0,0 +1,114 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using System.Linq;
+using Moq;
+using NUnit.Framework;
+using osu.Framework.Allocation;
+using osu.Framework.Bindables;
+using osu.Framework.Graphics;
+using osu.Framework.Graphics.UserInterface;
+using osu.Framework.Testing;
+using osu.Game.Screens.OnlinePlay.Multiplayer;
+using osu.Game.Screens.Play;
+using osuTK.Input;
+
+namespace osu.Game.Tests.Visual.Multiplayer
+{
+    public class TestSceneGameplayChatDisplay : MultiplayerTestScene
+    {
+        private GameplayChatDisplay chatDisplay;
+
+        [Cached(typeof(ILocalUserPlayInfo))]
+        private ILocalUserPlayInfo localUserInfo;
+
+        private readonly Bindable<bool> localUserPlaying = new Bindable<bool>();
+
+        private TextBox textBox => chatDisplay.ChildrenOfType<TextBox>().First();
+
+        public TestSceneGameplayChatDisplay()
+        {
+            var mockLocalUserInfo = new Mock<ILocalUserPlayInfo>();
+            mockLocalUserInfo.SetupGet(i => i.IsPlaying).Returns(localUserPlaying);
+
+            localUserInfo = mockLocalUserInfo.Object;
+        }
+
+        [SetUpSteps]
+        public override void SetUpSteps()
+        {
+            base.SetUpSteps();
+
+            AddStep("load chat display", () => Child = chatDisplay = new GameplayChatDisplay
+            {
+                Anchor = Anchor.Centre,
+                Origin = Anchor.Centre,
+                Width = 0.5f,
+            });
+
+            AddStep("expand", () => chatDisplay.Expanded.Value = true);
+        }
+
+        [Test]
+        public void TestCantClickWhenPlaying()
+        {
+            setLocalUserPlaying(true);
+
+            AddStep("attempt focus chat", () =>
+            {
+                InputManager.MoveMouseTo(textBox);
+                InputManager.Click(MouseButton.Left);
+            });
+
+            assertChatFocused(false);
+        }
+
+        [Test]
+        public void TestFocusDroppedWhenPlaying()
+        {
+            assertChatFocused(false);
+
+            AddStep("focus chat", () =>
+            {
+                InputManager.MoveMouseTo(textBox);
+                InputManager.Click(MouseButton.Left);
+            });
+
+            setLocalUserPlaying(true);
+            assertChatFocused(false);
+
+            // should still stay non-focused even after entering a new break section.
+            setLocalUserPlaying(false);
+            assertChatFocused(false);
+        }
+
+        [Test]
+        public void TestFocusOnTabKeyWhenExpanded()
+        {
+            assertChatFocused(false);
+            AddStep("press tab", () => InputManager.Key(Key.Tab));
+            assertChatFocused(true);
+        }
+
+        [Test]
+        public void TestFocusOnTabKeyWhenNotExpanded()
```

Comment:
Judging from the code, this needs a `setLocalUserPlaying(true)` at the start to pass.

Reply:
Weird, I guess I was always getting this from a previous step when testing.